added one login docs (#127)
common-fate/docs: docs/approvals/sso/onelogin.md

---
sidebar_position: 3
---

> # One Login

Reviewer: This should be styled as "OneLogin" (note this needs changing in a few places)

```suggestion
# OneLogin
```

:::info

> Granted Approvals user and group sync requires some credentials to be configured in your identity provider account. Granted Approvals uses 2LO authentication to read users and groups from your directory and sync them to an internal database every 5 minutes.

Reviewer: Can you add a link to 2LO authentication for OneLogin? I think it might be better to call this "OAuth2.0 authentication", I can't find "2LO authentication" when I Google it for OneLogin
```suggestion
Granted Approvals user and group sync requires some credentials to be configured in your identity provider account. Granted Approvals uses [OAuth2.0 authentication](https://developers.onelogin.com/api-docs/1/oauth20-tokens/generate-tokens-2) to read users and groups from your directory and sync them to an internal database every 5 minutes.
```

Not sure if that is the best link

:::

> ## One Login setup

Reviewer: ```suggestion
## Setup
```

### Creating Access Tokens in One Login

To set up One Login to sync users and groups with Granted Approvals we will need to create an access token to the developer API.

Sign in to your One Login admin portal using a user profile with admin privileges.

From the home screen hover over the **Developers** tab in the navbar, then from the options select **API Credentials**
![](/img/sso/one-login/1.png)

On the API Credentials screen there will be a button in the top right of the page to make a **New Credential**, click it to make a new credential.

![](/img/sso/one-login/2.png)

![](/img/sso/one-login/3.png)
- Give your credential a descriptive name
- Make sure *Read users* is the selected permission
Save/Create the credential.

A new modal will open, leave this open and move onto the next steps.

> ### Running Gdeploy Commands

Reviewer: `gdeploy` should always be stylised in lower case with code formatting around it

```suggestion
### Running `gdeploy` Commands
```

This is where we can start up the `gdeploy identity sso enable` command. Run the following to begin the SSO setup:

> ```json

Reviewer: ```suggestion
```bash
```

gdeploy identity sso enable
```

Select 'One Login' when prompted for the identity provider.

> ```json

Reviewer: ```suggestion
```bash
```

? The SSO provider to deploy with  [Use arrows to move, type to filter]
  Google
  Okta
> One Login
```

Head back to your One Login admin portal where you left it open, there will be a client ID and Client secret which you will be able to copy.

`gdeploy` will prompt you for IDs relating to your One Login tenancy.

> 1. For the `Base Url` param, get your tenancies url. This can be copied from your admin portal. It will have a format like this: `https://{tenantName}.onelogin.com`

Reviewer: Use parameter rather than param. Parameter is a lot more readable and isn't that much longer

```suggestion
1. For the `Base Url` parameter, get your tenancies url. This can be copied from your admin portal. It will have a format like this: `https://{tenantName}.onelogin.com`
```
